avrgirl-usbtinyisp.js

```diff
@@ -1,432 +1,414 @@
-
-  EventEmitter.call(this);
-
-  var self = this;
-  this.programmer.open(function(error) {
-    if (error) {
-      console.error(error);
-      return;
-    }
-    setImmediate(_emitReady, self);
-  });
-}
+const C = require('./lib/c');
+const util = require('util');
+const EventEmitter = require('events').EventEmitter;
+const usbtinyisp = require('usbtinyisp');
+const bufferEqual = require('buffer-equal');
+const async = require('async');
+const programmers = require('./lib/programmers');
+const intelhex = require('intel-hex');
+const fs = require('fs');
+const chips = require('avrgirl-chips-json');
+
+class avrgirlUsbTinyIsp extends EventEmitter {
+  /**
+   * Constructor
+   *
+   * @param {object} options - options for consumer to pass in
+   */
+  constructor(options) {
+    super();
+
+    var self = this;
+
+    // set up noisy or quiet
+    self.debug = options.debug ? console.log.bind(console) : () => {};
+
+    // most people won't need this level of debug output
+    self.hackerLog = options.hackerMode ? console.log.bind(console) : () => {};
+
+    self.options = {
+      sck: options.sck || C.SCK_DEFAULT,
+      debug: options.debug || false,
+      chip: options.chip || chips.attiny85,
+      log: self.hackerLog,  // for usbtinyisp lib
+      programmer: options.programmer || null
+    };
+
+    // fix this pls self, it's very unattractive
+    // do an error check too
+    var p = self.options.programmer ? programmers[self.options.programmer] : null;
+    self.options.pid = p ? p.pid : 3231;
+    self.options.vid = p ? p.vid : 6017;
+
+    // create new instance of usbtiny isp as programmer instance
+    self.programmer = new usbtinyisp(self.options);
+
+    self.programmer.open(error => {
+      if (error) {
+        console.error(error);
+        return;
+      }
+      setImmediate(_emitReady, self);
+    });
+  }
 
-util.inherits(avrgirlUsbTinyIsp, EventEmitter);
+  /**
+   * Primes the programmer and the microchip for programming
+   * Sets the clock speed of the programmer, and enables programming mode on the chip
+   *
+   * @param {function} callback - function to run upon completion/error
+   */
+  enterProgrammingMode(callback) {
+    var self = this;
 
-// ready event emitter
-function _emitReady (self) {
-  self.emit('ready');
-}
+    var cmd = new Buffer(self.options.chip.pgmEnable);
 
-/**
- * Primes the programmer and the microchip for programming
- * Sets the clock speed of the programmer, and enables programming mode on the chip
- *
- * @param {function} callback - function to run upon completion/error
- */
-avrgirlUsbTinyIsp.prototype.enterProgrammingMode = function (callback) {
-  var self = this;
-  var cmd = new Buffer(this.options.chip.pgmEnable);
-
-  this.setSCK(this.options.sck, function(error) {
-    if (error) { return callback(error) }
-    // is this timeout necessary?
-    setTimeout(function() {
-      self.programmer.spi(cmd, function (error) {
+    self.setSCK(self.options.sck, error => {
```

Comment:
There were a couple places where `var self = this;` pattern is used, but `self` was not referenced consistently in the body of the function. Right here I changed `this` to `self` to be more consistent.